src/components/drawer/drawer.js

```diff
@@ -0,0 +1,162 @@
+
+import '@brightspace-ui/core/components/button/button-icon.js';
+
+import { drawerFooterStyles, drawerHeaderStyles, drawerStyles } from './drawer-styles.js';
+import { html, LitElement, nothing } from 'lit';
+import { classMap } from 'lit/directives/class-map.js';
+import { DrawerMixin } from './drawer-mixin.js';
+import { getUniqueId } from '@brightspace-ui/core/helpers/uniqueId.js';
+import { heading3Styles } from '@brightspace-ui/core/components/typography/styles.js';
+import { ifDefined } from 'lit/directives/if-defined.js';
+import { LocalizeLabsElement } from '../../../helpers/localizeLabsElement.js';
+
+const mediaQueryList = window.matchMedia('(max-width: 615px), (max-height: 420px) and (max-width: 900px)');
+
+/**
+ * Modal element that is anchored to one side of the viewport or parent element.
+ * @fires d2l-drawer-before-close - Dispatched with bound action value before the dialog is closed for any reason, providing an opportunity to prevent the closing.
+ * @slot - Default slot for content inside drawer
+ * @slot footer - Slot for footer content such as workflow buttons. Apply attribute `drawer-close` to workflow buttons to have them automatically close the drawer with the action's bound `value`.
+ */
+class Drawer extends LocalizeLabsElement(DrawerMixin(LitElement)) {
```

Comment:
Is the mixin even needed? I know this was based off dialog, but it's using a mixin to share behaviour across several components.

Reply:
I was at the stage where I wasn't sure what the future of this component would hold and didn't know if there would ever be fracturing along use case lines - so I figured I would keep as a mixin for now and fold it in if it wasn't needed